### PR TITLE
Add pwd and -h option for grep in DWAINE

### PR DESCRIPTION
**Pull request by system86**
## About the PR
Adds pwd, a utility for **p**rinting your current **w**orking **d**irectory
Adds -h option to grep, which omits the "filename:linenumber:" prefix on matched lines
also apparently slightly randomizes the number of spam emails you get, to reward the 2 people that even read the damn things less or more depending on how RNGesus is feeling at the time

## Why's this needed?
for pwd: to allow checking/logging a user's current working directory
for grep -h: to allow even nerdier nerd shit, namely processing (making readable) large amounts of captured PDA messages, almost like you have access to a multi-million-credit state-of-the-art computer system or something. having filenames and stuff forcefully appended to everything makes grep not very useful

### CODE REVIEW

File: code/modules/networks/computer3/mainframe2/utilities.dm [hunk 1]
```diff
@@ -1402,3 +1408,10 @@
 
 		mainframe_prog_exit
 		return
+/datum/computer/file/mainframe_program/utility/pwd
+	name = "pwd"
+	size = 1
+	initialize(var/initparams)
+		message_user(read_user_field("curpath"))
+		mainframe_prog_exit
+	return
```
Comment (ZeWaka) [inline code/modules/networks/computer3/mainframe2/utilities.dm:1417]: not needed / won't work / might not compile

Comment (system86) [reply code/modules/networks/computer3/mainframe2/utilities.dm:1417]: oof. will fix

Comment (system86) [reply code/modules/networks/computer3/mainframe2/utilities.dm]: right, i've got rid of the random return

**Comment by ZeWaka** [comment]
Yo @system86, this is causing runtimes and then a infinite loop -> crash
```
[00:03:35] ,: Regex fail: ? + * or { follows nothing
proc name: New (/regex/New)
  usr: Diego Garland (/mob/living/carbon/human)
  src: /regex (/regex)
  usr.loc: the steel plating (173,105,1) (/turf/simulated/floor/plating)
  call stack:
/regex (/regex): New("*", null)
grep (/datum/computer/file/mainframe_program/utility/grep): initialize("-hr * .")
Mainframe (/obj/machinery/networked/mainframe/zeta): run program(grep (/datum/computer/file/mainframe_program/utility/grep), /datum/mainframe2_user_data (/datum/mainframe2_user_data), Msh (/datum/computer/file/mainframe_program/shell), "-hr * .", 0)
Kernel (/datum/computer/file/mainframe_program/os/kernel): receive progsignal(160, /list (/list), null)
Mainframe (/obj/machinery/networked/mainframe/zeta): relay progsignal(Msh (/datum/computer/file/mainframe_program/shell), 1, /list (/list), null)
Msh (/datum/computer/file/mainframe_program/shell): signal program(1, /list (/list), null)
Msh (/datum/computer/file/mainframe_program/shell): execpath("/bin/grep", "/mnt/radio/1149/", "grep", /list (/list), 0)
Msh (/datum/computer/file/mainframe_program/shell): input text("grep -hr * .")
Kernel (/datum/computer/file/mainframe_program/os/kernel): term input("grep -hr * .", "02025510", null, null)
Mainframe (/obj/machinery/networked/mainframe/zeta): receive signal(/datum/signal (/datum/signal))
...
Terminal card (/obj/item/peripheral/network/powernet_card/terminal): receive command(DWAINE Terminal (/obj/machinery/computer3/terminal/zeta), "transmit", /datum/signal (/datum/signal))
DWAINE Terminal (/obj/machinery/computer3/terminal/zeta): send command("transmit", /datum/signal (/datum/signal), "\[0x2025510]")
TermOS B (/datum/computer/file/terminal_program/os/terminal_os): peripheral command("transmit", /datum/signal (/datum/signal), "\[0x2025510]")
TermOS B (/datum/computer/file/terminal_program/os/terminal_os): send term message("grep -hr * .", 0)
TermOS B (/datum/computer/file/terminal_program/os/terminal_os): input text("grep -hr * .")
DWAINE Terminal (/obj/machinery/computer3/terminal/zeta): Topic("src=\[0x20157ae]&command=grep%...", /list (/list))
DesuDesuDesuDesuDesuDesuDesuDe (/client): Topic("src=\[0x20157ae]&command=grep%...", /list (/list), DWAINE Terminal (/obj/machinery/computer3/terminal/zeta))
DesuDesuDesuDesuDesuDesuDesuDe (/client): Topic("src=\[0x20157ae]&command=grep%...", /list (/list), DWAINE Terminal (/obj/machinery/computer3/terminal/zeta))
DesuDesuDesuDesuDesuDesuDesuDe (/client): Topic("src=\[0x20157ae]&command=grep%...", /list (/list), DWAINE Terminal (/obj/machinery/computer3/terminal/zeta))
```

**Comment by system86** [comment]
grep seems to be bugging out even without my spaghetti code in the mix. i'm opening an issue

**Comment by system86** [comment]
putting "*" in grep seems to break everything. someone made grep run on actual regexes instead of the weak shit they ran on circa 2019. * used to be a wildcard, now it crashes. W E I R D.

**Comment by ZeWaka** [comment]
> putting "*" in grep seems to break everything. someone made grep run on actual regexes instead of the weak shit they ran on circa 2019. * used to be a wildcard, now it crashes. W E I R D.

So, it wasn't due to the contents of the patch?

**Comment by system86** [comment]
Doesn't seem to be. I did a whole bunch of greps, only some of them involved *, and only those *seem* to be causing the runtime errors.

**Comment by system86** [comment]
"*" is a special character in regex, so maybe having a regex with JUST it causes issues.

**Pull request merged**